### DISCUSSION TITLE
v0.1.0

### DISCUSSION THREAD

**Discussion by carson-katri** [carson-katri/dream-textures#566] [Announcements]
This update has [so](https://github.com/carson-katri/dream-textures/pull/435) [many](https://github.com/carson-katri/dream-textures/pull/484) [bug](https://github.com/carson-katri/dream-textures/pull/485) [fixes](https://github.com/carson-katri/dream-textures/pull/479). It also includes AMD support, seamless axis detection, projection baking, optimizations and UI enhancements, and more.

Join the [Dream Textures Discord server](https://discord.gg/EmDJ8CaWZ7) to discuss the add-on, share your art, and get help from other users.

## Choose Your Installation
Several versions are available. Find the one that will work for you.

<details>
<summary>Windows</summary>

### NVIDIA GPU
* [Dream Textures for Windows \(CUDA\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-windows-cuda.7z) - Local and cloud rendering with NVIDIA CUDA
    * Due to file size limits on GitHub, the add-on ZIP file is contained in a 7-Zip archive.
    * Install [7-Zip](https://www.7-zip.org/), right click on the downloaded `dream_textures-windows-cuda.7z` archive, select "7-Zip" > "Extract Here" to get the `dream_textures-windows-cuda.zip` file.
    * In Blender preferences open "Add-ons" > "Install...", then choose the `dream_textures-windows-cuda.zip` file.
* [Dream Textures for DreamStudio \(Windows\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-windows-dreamstudio.zip) - Cloud rendering with DreamStudio

### AMD GPU
* [Dream Textures for Windows \(DirectML\)]() - Local and cloud rendering with DirectML
* [Dream Textures for DreamStudio \(Windows\)](https://github.com/carson-katri/dream-textures/releases/download/0.1.0/dream_textures-windows-dreamstudio.zip) - Cloud rendering with DreamStudio
</details>

<details>
<summary>macOS</summary>

### Apple Silicon (M1/M2)
* [Dream Textures for Apple Silicon](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-macos-arm.zip) - Local and cloud rendering for Apple Silicon Macs
* [Dream Textures for DreamStudio \(Apple Silicon\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-macos-arm-dreamstudio.zip) - Cloud rendering with DreamStudio

### Intel
* [Dream Textures for DreamStudio \(Intel\)](https://github.com/carson-katri/dream-textures/releases/download/0.0.9/dream_textures-macos-intel-dreamstudio.zip) - Cloud rendering with DreamStudio
> *Note* Local generation is not available for Intel Macs.
</details>

<details>
<summary>Linux</summary>

Installation must be completed manually. Follow the instructions to [install manually](https://github.com/carson-katri/dream-textures/blob/main/docs/DEVELOPMENT_ENVIRONMENT.md) for more details.
</details>

<details>
<summary>
<a href="https://www.blendermarket.com/products/dream-textures">Blender Market</a>
</summary>

You can optionally purchase it for a small fee on Blender Market.

> The version on Blender Market is identical to the free download available here. It is simply a way to support my work on this tool.

</details>

## Setup
After installing the appropriate add-on ZIP file, enable the add-on and expand its preferences. There you will find further instructions for setup.

See the [setup guide](https://github.com/carson-katri/dream-textures/blob/main/docs/SETUP.md) for more detailed instructions.

### Automatic Seamless Detection
The axes the open image is seamless along is automatically detected for better upscaling and image to image results.

![](https://user-images.githubusercontent.com/47096043/209588718-f0a25e66-c603-4543-b73e-4b97df225042.gif)

### Batch Size Optimization
Increase the batch size for significantly faster generation of multiple iterations, file batch, or upscaling.

### Automatic Projection Baking / Better UV handling
Enable the "Bake" option to project back onto the mesh's original UV map.

Dream Textures will now preserve your original UV map instead of overwriting it.

<img width="261" alt="Screenshot_2023-01-01_at_6 50 30_PM" src="https://user-images.githubusercontent.com/13581484/210187825-91349f1e-5d27-474d-8b2d-95fe20207ecb.png">
<img width="303" alt="Screenshot 2023-01-01 at 6 53 38 PM" src="https://user-images.githubusercontent.com/13581484/210187832-e5a1169d-baeb-49c0-a3f8-8a8efa57e8a2.png">
<img width="307" alt="Screenshot_2023-01-01_at_6 50 44_PM" src="https://user-images.githubusercontent.com/13581484/210187826-6914a62e-161a-43de-89e6-2ff5389feb56.png">

### AMD Support via DirectML
All DirectX 12-capable GPUs are now compatible with Dream Textures! Download the `directml` version of the add-on if you have a non-NVIDIA GPU.

### UI Improvements
Image names now include the prompt alongside the seed.

![](https://user-images.githubusercontent.com/6715569/212480061-794470e7-886e-40e1-b2ef-ecc2b82f5ebb.png)

Many common issues are now caught before you ever press "Generate".

<img width="300" src="https://user-images.githubusercontent.com/13581484/213805771-b4abe4da-9d63-4124-9e75-18a7e6be89cc.png" />

Size options are now disabled by default, and will use the model's preferred size.
<img width="300" src="https://user-images.githubusercontent.com/13581484/213805815-d4a41052-f60f-4d18-9949-e3133b851b4c.png" />

## Full Changelog
* Always use spawn multiprocessing context by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/435
* Stop reusing the first UV map slot for texture projection by @Dekker3D in https://github.com/carson-katri/dream-textures/pull/480
* Automatic Seamless Detection by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/410
* CUDA fixes by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/484
* Fix Iterations and File Batch by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/485
* Batch Size Optimization by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/486
* Automatic Projection Baking by @carson-katri in https://github.com/carson-katri/dream-textures/pull/495
* Properly handle `fp16` branches by @carson-katri in https://github.com/carson-katri/dream-textures/pull/479
* Fix link in Doc by @playerla in https://github.com/carson-katri/dream-textures/pull/521
* Add Readme Troubleshooting Tips by @JasonHoku in https://github.com/carson-katri/dream-textures/pull/519
* DirectML Execution by @NullSenseStudio in https://github.com/carson-katri/dream-textures/pull/507
* Add Prompt String To Material And Texture Names by @JasonHoku in https://github.com/carson-katri/dream-textures/pull/499
* Add DreamPrompt validation by @carson-katri in https://github.com/carson-katri/dream-textures/pull/530
* Release v0.1.0 by @carson-katri in https://github.com/carson-katri/dream-textures/pull/551

## New Contributors
* @playerla made their first contribution in https://github.com/carson-katri/dream-textures/pull/521
* @JasonHoku made their first contribution in https://github.com/carson-katri/dream-textures/pull/519

**Full Changelog**: https://github.com/carson-katri/dream-textures/compare/0.0.9...0.1.0

<hr /><em>This discussion was created from the release <a href='https://github.com/carson-katri/dream-textures/releases/tag/0.1.0'>v0.1.0</a>.</em>

**Comment by CriticalGraphics** [comment]
This is Awesome! Thanks, just about to test!

**Comment by faxlab** [comment]
Super excited

**Comment by THEAETIK** [comment]
"Dream Textures will now preserve your original UV map instead of overwriting it."

While this is a neat feature to have, this is rather unintuitive at the moment IMO. By default there's always much retouching needed on the UVs and having the projection baked on the first attempt seems unproductive. Or I'm missing the point.

I really wish there was still an option for project from camera.

**Reply by carson-katri** [reply]
That’s still the default. But it creates a new UV map for the ones projected from the camera angle instead of updating the first UV map. It will only use that original UV map if you enable “Bake”

**Comment by AuraNETinfo** [comment]
Traceback (most recent call last):
  File "C:\Program Files\Blender Foundation\Blender 3.4\3.4\scripts\modules\addon_utils.py", line 369, in enable
    mod.register()
  File "C:\Users\name\AppData\Roaming\Blender Foundation\Blender\3.4\scripts\addons\dream_textures\__init__.py", line 64, in register
    raise RuntimeError("Another instance of Dream Textures is already running.")
RuntimeError: Another instance of Dream Textures is already running.
